plugins/catalog-backend/src/database/operations/stitcher/markForStitching.ts

```diff
@@ -36,41 +37,65 @@ export async function markForStitching(options: {
   const entityIds = split(options.entityIds);
   const knex = options.knex;
 
-  for (const chunk of entityRefs) {
-    await knex
-      .table<DbFinalEntitiesRow>('final_entities')
-      .update({
-        hash: 'force-stitching',
-      })
-      .whereIn(
-        'entity_id',
-        knex<DbRefreshStateRow>('refresh_state')
-          .select('entity_id')
-          .whereIn('entity_ref', chunk),
-      );
-    await knex
-      .table<DbRefreshStateRow>('refresh_state')
-      .update({
-        result_hash: 'force-stitching',
-        next_update_at: knex.fn.now(),
-      })
-      .whereIn('entity_ref', chunk);
-  }
+  if (options.strategy.mode === 'immediate') {
+    for (const chunk of entityRefs) {
+      await knex
+        .table<DbFinalEntitiesRow>('final_entities')
+        .update({
+          hash: 'force-stitching',
+        })
+        .whereIn(
+          'entity_id',
+          knex<DbRefreshStateRow>('refresh_state')
+            .select('entity_id')
+            .whereIn('entity_ref', chunk),
+        );
+      await knex
+        .table<DbRefreshStateRow>('refresh_state')
+        .update({
+          result_hash: 'force-stitching',
+          next_update_at: knex.fn.now(),
+        })
+        .whereIn('entity_ref', chunk);
+    }
+
+    for (const chunk of entityIds) {
+      await knex
+        .table<DbFinalEntitiesRow>('final_entities')
+        .update({
+          hash: 'force-stitching',
+        })
+        .whereIn('entity_id', chunk);
+      await knex
+        .table<DbRefreshStateRow>('refresh_state')
+        .update({
+          result_hash: 'force-stitching',
+          next_update_at: knex.fn.now(),
+        })
+        .whereIn('entity_id', chunk);
+    }
+  } else {
```

Comment:
in theory you could put anything other than immediate, not sure that's a problem but I'd prefer explicit over implicit (especially when the branch is long so I have to scroll up to see what the initial statement was)

Reply:
Sure, made this into an else-if and throwing if it was an unknown mode